src/ast/field_analyser.cpp

```diff
@@ -216,22 +229,59 @@ void FieldAnalyser::visit(Predicate &pred)
   pred.expr->accept(*this);
 }
 
-void FieldAnalyser::visit(AttachPoint &ap __attribute__((unused)))
+bool FieldAnalyser::compare_args(std::map<std::string, SizedType> args1,
+                                 std::map<std::string, SizedType> args2)
```

Comment:
I think we can save a copy here
```suggestion
bool FieldAnalyser::compare_args(const std::map<std::string, SizedType>& args1,
                                 const std::map<std::string, SizedType>& args2)
```

Reply:
true, will change